src/theory/theory_model_builder.cpp

```diff
@@ -935,6 +937,11 @@ bool TheoryEngineModelBuilder::buildModel(TheoryModel* tm)
       {
         i2 = i;
         ++i;
+        if (evaluableEqc.find(*i2) != evaluableEqc.end())
+        {
+          // we never assign to evaluable equivalence classes
+          continue;
```

Comment:
What if all equivalence classes have evaluable terms in them?  Like the example in my comment on line 1065?

Reply:
My impression is that this should never happen, or if it does, we should just fail to construct the model.

Being more pedantic here I think simplifies things upstream, and was the reason I changed the behavior of NL (see other comment)..

Reply:
I think this does happen in the small example in the issue.  There are two equivalence classes.  Both have evaluable and assignable objects.  I think we need to think carefully about how to handle this case.